src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,178 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel --------------------
+// Single channel resample support
+__global__ void resample_1channel_hip_tensor(float *srcPtr,
+                                             float *dstPtr,
+                                             float inRateTensor,
+                                             float outRateTensor,
+                                             int srcLength,
+                                             int outEnd,
+                                             double scale,
+                                             RpptResamplingWindow &window)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+
+    int block = 256;
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
```

Comment:
Since you are processing block number of elements in each thread
The exit condition i think should be 


```    
if (outBlock >= outEnd / block)
            return;
```
Please check this and make this change if correct

Reply:
A block might contain values greater than outEnd too towards end thus added this kind of check instead of (outEnd / block)

Reply:
Again dividing by /block will just increase complexity.

Reply:
You can use ceil(outEnd/block) if that is the case

Reply:
Discussed over call and explained